parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -518,7 +519,9 @@ pub mod pallet {
 				});
 			};
 
-			if update.finalized_header.slot > latest_finalized_state.slot {
+			if update.finalized_header.slot > latest_finalized_state.slot &&
+				!update.block_roots_branch.is_empty()
+			{
```

Comment:
If we don't store the finalized header if the `block_roots_branch` is empty, does it makes sense for the relayer to send this update in the first place? Would it not be a pointless update then?